system/web/context/RequestContext.cfc

```diff
@@ -230,461 +230,553 @@ component serializable=false accessors="true"{
-	*
-	* @return RequestContext
-	*/
+	 * Just like cfparam, but for the private request collection
+	 * @name The key name
+	 * @value The value
+	 *
+	 * @return RequestContext
+	 */
 	function paramPrivateValue( required name, required value ){
 		arguments.private = true;
-		return paramValue( argumentCollection=arguments );
+		return paramValue( argumentCollection = arguments );
 	}
 
 	/************************************** CURRENT CONTEXT METHODS *********************************************/
 
 	/**
-	* Gets the current set view the framework will try to render for this request
-	*/
+	 * Gets the current set view the framework will try to render for this request
+	 */
 	string function getCurrentView(){
 		return getPrivateValue( "currentView", "" );
 	}
 
 	/**
-	* Gets the current set view the framework will try to render for this request
-	*/
+	 * Gets the current set view the framework will try to render for this request
+	 */
 	struct function getCurrentViewArgs(){
 		return getPrivateValue( "currentViewArgs", structNew() );
 	}
 
 	/**
-	* Gets the current set views's module for rendering
-	*/
+	 * Gets the current set views's module for rendering
+	 */
 	string function getCurrentViewModule(){
 		return getPrivateValue( "viewModule", "" );
 	}
 
 	/**
-	* Gets the current set layout for rendering
-	*/
+	 * Gets the current set layout for rendering
+	 */
 	string function getCurrentLayout(){
 		return getPrivateValue( "currentLayout", "" );
 	}
 
 	/**
-	* Gets the current set layout's module for rendering
-	*/
+	 * Gets the current set layout's module for rendering
+	 */
 	string function getCurrentLayoutModule(){
 		return getPrivateValue( "layoutmodule", "" );
 	}
 
 	/**
-	* Get the current request's SES route that matched
-	*/
+	 * Get the current request's SES route that matched
+	 */
 	string function getCurrentRoute(){
 		return getPrivateValue( "currentRoute", "" );
 	}
 
 	/**
-	* Get the current request's SES route name
-	*/
+	 * Get the current request's SES route name
+	 */
 	string function getCurrentRouteName(){
 		return getPrivateValue( "currentRouteName", "" );
 	}
 
 
 	/**
-	* Get the current routed URL that matched the SES route
-	*/
+	 * Get the current routed URL that matched the SES route
+	 */
 	string function getCurrentRoutedURL(){
 		return getPrivateValue( "currentRoutedURL", "" );
 	}
 
 	/**
-	* Get the current routed namespace that matched the SES route, if any
-	*/
+	 * Tests that a given path exists in the currently routed URL.
+	 * If the exact flag is passed, the path must match exactly.
+	 * Using the `urlMatchesExact` is preferred.
+	 *
+	 * @path    The path to match in the currently routed URL.
+	 * @exact   Flag to do an exact match instead of a partial match
+	 *
+	 * @return  Boolean
+	 */
+	boolean function urlMatches( required string path, boolean exact = false ){
+		var currentRoutedURL = getCurrentRoutedURL();
+		var replaced         = replace( currentRoutedURL, arguments.path, "" );
```

Comment:
Why do this if you are looking for an exact match, why not just do

```
if( arguments.exact ){
return arguments.path == currentRoutedUrl;
}
```
This way you avoid setting any state.

Reply:
Updated